sqldelight-gradle-plugin/src/test/fixtures/well-formed-selects/expected/com/sample/TeamModel.java

```diff
@@ -138,4 +135,19 @@ public T won_cup(boolean won_cup) {
       return (T) this;
     }
   }
+
+  final class Factory<T extends TeamModel> {
```

Comment:
hmmm not sure. the only benefit the current way has is its semantically identical to how other queries are handled